src/CozyUtils.spec.js

```diff
@@ -241,6 +241,57 @@ describe('CozyUtils', () => {
       expect(result).toEqual(fakeAccount)
     })
 
+    it('should update an existing account with the same email', async () => {
+      const previousAccount = {
+        _id: '123',
+        _rev: 'abc',
+        canLinkContacts: true,
+        shouldSyncOrphan: true,
+        lastSync: null,
+        lastLocalSync: null,
+        name: 'john.doe@gmail.com',
+        _type: DOCTYPE_CONTACTS_ACCOUNT,
+        type: APP_NAME,
+        sourceAccount: 'previousAccountId',
+        version: 1
+      }
+      const findSpy = jest.fn().mockResolvedValueOnce({
```

Comment:
maybe add a comment to explain this : `we search the accounts using the new sourceAccount` or something? (maybe also add a comment in the code that explains why/in which case we search by email if we don't find by `sourceAccount`?)